### ISSUE TITLE
Document how to use Prisma in monorepos

### ISSUE THREAD

**Issue by nikolasburk** [prisma/web#1288]
e.g. using:

- Yarn workspaces
- Lerna
- ngx

Note that there's a simple example with Yarn workspaces here: https://github.com/nikolasburk/monorepo-prisma

More context on the internal [Slack](https://prisma-company.slack.com/archives/C5Z9TH6N9/p1613033318151900):

<details><summary>screenshot</summary>

![image](https://user-images.githubusercontent.com/4058327/107616595-ab2c7b00-6c4e-11eb-86a0-97c19dad155a.png)

</details>

**Comment by johannesschobel** [comment]
I would like to add `nrwl/nx` ( https://github.com/nrwl/nx ) to the list of monorepos.

**Comment by tsongas** [comment]
Vercel just bought [Turborepo](https://turborepo.org/) I think this is going to become very important very quickly!

**Comment by commenda-main** [comment]
The example at https://github.com/nikolasburk/monorepo-prisma was helpful for figuring out how to use prisma client, but when I try to use prisma types I'm hitting issues. I'd appreciate documentation on how to export types from the module as well! 

I naively tried to 
```
export * from '@prisma/client'
```
in my index.ts, but hitting issues with next.js static page generation timeout.

**Comment by yorch** [comment]
From what I've been testing, doesn't actually work when using ENV variables unless I place the `.env` file in the prisma directory.

So for instance, I have this monorepo with yarn workspaces
```
 |- packages/
     |- server/
         |- package.json
 |- prisma/
     |- schema.prisma
     |- package.json
 |- package.json
 |- .env
```

Seems like it needs to have the `.env` file in side the `prisma` directory instead of reading the file from the file on root

**Comment by Liam-Tait** [comment]
Related https://github.com/vercel/turborepo/pull/979#issuecomment-1183444069

**Comment by zomars** [comment]
Yes please, also declaring `.env` files on the root of the monorepo break prisma `env()` function if prisma schema is not located at `prisma/schema.prisma`

**Comment by zomars** [comment]
> From what I've been testing, doesn't actually work when using ENV variables unless I place the `.env` file in the prisma directory.
> 
> So for instance, I have this monorepo with yarn workspaces
> 
> ```
>  |- packages/
>      |- server/
>          |- package.json
>  |- prisma/
>      |- schema.prisma
>      |- package.json
>  |- package.json
>  |- .env
> ```
> 
> Seems like it needs to have the `.env` file in side the `prisma` directory instead of reading the file from the file on root

Yes! +1 to this 🙌🏽 

EDIT: Relevant issue as well prisma/prisma#12535

**Comment by leerob** [comment]
Related example: https://github.com/vercel/turborepo/tree/main/examples/with-prisma

**Comment by kokokenada** [comment]
I have an issue with monorepo using npm workspaces and prisma.  The set up is there are two microservices that both use prisma (say packages/api1, packages/api2).  If the same version of prisma is used in each, npm will create @prisma in the root node_modules.   The generated .prisma files are installed as a sibling to @prisma, so api1 and api2 can clobber each other.  One can specify where to generate the files, however, other dependencies (i.e. nestjs) uses @prisma/client to import, so the new location is ignored.

**Comment by nikolasburk** [comment]
> however, other dependencies (i.e. nestjs) uses @prisma/client to import, so the new location is ignored.

@kokokenada Do you think this could be configurable in your NestJS app? If no, it might be worth opening an issue in the [`nest`](https://github.com/nestjs/nest) repo or submit a PR to add this option.

**Comment by kokokenada** [comment]
Thanks for the reply @nikolasburk, and yes, I also opened a ticket there: https://github.com/nestjs/nest/issues/10150 and moved to https://github.com/notiz-dev/nestjs-prisma/issues/31

**Comment by skworden** [comment]
I have the same issue as @kokoenada . I can't have a turborepo monorepo with two different Prisma schemas - Prisma gets confused about which schema to use.

**Comment by sebpowell** [comment]
I came across this thread whilst looking for a solution to this: https://github.com/vercel/turbo/discussions/5610

Wondering if anyone has experienced a similar issue?

**Edited:** found the fix, see here in case it's of use to anyone: https://www.prisma.io/docs/guides/other/troubleshooting-orm/help-articles/nextjs-prisma-client-monorepo

**Comment by antonio-ivanovski** [comment]
For my case where the monorepo will have multiple packages with different schemas, I opted to generate separate prisma clients into separate directories.


```prisma
// packages/accounts/prisma.schema

generator client {
  provider = "prisma-client-js"
  output   = "./node_modules/@prisma/accounts-client" // <-- note this
  // output   = "./node_modules/@prisma/dashboard-client" // <-- for other package "dashboard"
}
```

```ts
// pacakges/accounts/src/...
import { PrismaClient } from '@prisma/accounts-client';

// ...
```

Can anyone tell me if this is OK approach or I will have some unexpected issues along the way?

**Comment by jharrell** [comment]
Hi all! We now have a guide on [integrating Prisma ORM with Turborepo](https://www.prisma.io/docs/guides/using-prisma-orm-with-turborepo). We're also listening to our community on what other tools we should create docs on and will be planning those!

Thanks everyone for chiming in here and I hope you like the guide 😊

**Comment by jason-daitrix** [comment]
> Hi all! We now have a guide on [integrating Prisma ORM with Turborepo](https://www.prisma.io/docs/guides/using-prisma-orm-with-turborepo). We're also listening to our community on what other tools we should create docs on and will be planning those!
> 
> Thanks everyone for chiming in here and I hope you like the guide 😊

The link is 404 now.

**Comment by jharrell** [comment]
@jason-daitrix thanks for the callout! Link fixed.
